### PR TITLE
refact: use hooks in more places + eliminate projects double-fetch

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

**Review by v-rocheleau — CHANGES_REQUESTED** [review]
small stuff, looks good

File: src/modules/metadata/actions.js [hunk 1]
```diff
@@ -65,19 +66,25 @@ const fetchProjects = networkAction(() => (_dispatch, getState) => ({
...
-
-  if (!state.services.itemsByKind.metadata) return Promise.resolve();
-  if (state.projects.isFetching || state.projects.isCreating || state.projects.isDeleting || state.projects.isSaving)
-    return Promise.resolve();
-
+  if (!getState().services.itemsByKind.metadata) return Promise.resolve();
   return dispatch(fetchProjects());
 };
 
+export const invalidateProjects = basicAction(INVALIDATE_PROJECTS);
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/modules/metadata/actions.js:86]: it means the data in the state is invalid and so should not be counted as "already fetched" for the purposes of dispatching a fetch action. i'll make a note.

File: src/components/manager/access/GrantForm.tsx [hunk 2]
```diff
@@ -72,7 +72,7 @@ const handleSubjectChange = (
 const SubjectInput = ({ value, onChange }: SubjectInputProps) => {
   const groups: StoredGroup[] = useGroups().data;
 
-  const homeIssuer = useOpenIdConfig()?.issuer ?? "";
+  const homeIssuer = useOpenIdConfig().data?.issuer ?? "";
```
[the remark this answers is not in the document]
Comment (davidlougheed) [reply src/components/manager/access/GrantForm.tsx:75]: data may be undefined but `useOpenIdConfig` cannot return undefined; fixed line 586

**Review by v-rocheleau — APPROVED** [review]
LGTM!

**Pull request merged**